openedx/core/djangoapps/content/course_metadata/tasks.py

```diff
@@ -0,0 +1,33 @@
+"""
+This module has implementation of celeary tasks to compute course aggregate metadata
+"""
+
+import logging
+
+from celery.task import task
+from opaque_keys.edx.keys import CourseKey
+from .models import CourseAggregatedMetaData
+from api_manager.courseware_access import get_course_leaf_nodes
+
+log = logging.getLogger('edx.celery.task')
+
+
+@task(name=u'lms.djangoapps.api_manager.tasks.update_course_metadata')
+def update_course_aggregate_metadata(course_key):
+    """
+    Regenerates and updates the course aggregate metadata (in the database) for the specified course.
+    """
+    if not isinstance(course_key, basestring):
+        raise ValueError('course_key must be a string. {} is not acceptable.'.format(type(course_key)))
+
+    course_key = CourseKey.from_string(course_key)
+
+    try:
+        assessments = get_course_leaf_nodes(course_key)
```

Comment:
or better, we could call it 'course_leaf_nodes" so down below we would see that total_assessments are equivalent to "course_leaf_nodes", at least for now